hr_timesheet_sheet/tests/test_hr_timesheet_sheet.py

```diff
@@ -608,3 +608,40 @@ def test_12_creating_sheet(self):
         self.assertEqual(line.unit_amount, 2.0)
         self.assertEqual(len(sheet.timesheet_ids), 1)
         self.assertEqual(len(sheet.line_ids), 7)
+
+    def test_13_analytic_account_multicompany(self):
+        new_employee = self.employee_model.create({
+            'name': "Test New Employee",
+            'user_id': self.user_2.id,
+            'company_id': self.company_2.id,
+        })
+        sheet = self.sheet_model.sudo(self.user_2).create({
+            'employee_id': new_employee.id,
+            'date_start': self.sheet_model._default_date_start(),
+            'date_end': self.sheet_model._default_date_end(),
+        })
+        self.assertEqual(sheet.company_id, self.company_2)
+
+        with self.assertRaises(UserError):
```

Comment:
And how is it raised this?

In the aal below, the only thing that has company 1 is the self.project_1, but the `_check_sheet_company_id` doesn't check the company of the project :S

Reply:
The project [inherits](https://github.com/odoo/odoo/blob/11.0/addons/project/models/project.py#L50) from the analytic account, so the company of the analytic account is the company of the project.

When you create an analytic line, [its company](https://github.com/odoo/odoo/blob/11.0/addons/analytic/models/analytic_account.py#L110) is the one of the analytic account.

Reply:
Oh, I understand now. Because the `readonly`, this case is not caught in the api.constrains.